src/Auth0.php

```diff
@@ -665,15 +664,17 @@ public function setAccessToken($accessToken)
      */
     public function setIdToken($idToken)
     {
-        $jwtVerifier          = new JWTVerifier([
-            'valid_audiences' => ! empty($this->idTokenAud) ? $this->idTokenAud : [ $this->clientId ],
-            'supported_algs' => $this->idTokenAlg ? [ $this->idTokenAlg ] : [ 'HS256', 'RS256' ],
-            'authorized_iss' => $this->idTokenIss ? $this->idTokenIss : [ 'https://'.$this->domain.'/' ],
-            'client_secret' => $this->clientSecret,
-            'secret_base64_encoded' => $this->clientSecretEncoded,
-            'guzzle_options' => $this->guzzleOptions,
-        ]);
-        $this->idTokenDecoded = (array) $jwtVerifier->verifyAndDecode( $idToken );
+        $sigVerifier = null;
+        if ('RS256' === $this->idTokenAlg) {
+            $jwksFetcher = new JWKFetcher($this->cacheHandler, $this->guzzleOptions);
+            $jwks        = $jwksFetcher->getKeys($this->idTokenIss.'.well-known/jwks.json');
+            $sigVerifier = new JwksVerifier( $jwks );
```

Comment:
> how does sound renaming `JwksVerifier` to `AsymmetricVerifier`?

I like it **much, much** better and was going to do that in this PR but didn't want to stray from the example :)

> put the JWKS logic and cache logic inside that class rather than here, as they belong to the verifier.

I disagree. This should accept the key it needs, not deal with HTTP requests and caching. Testing is easier, concerns isolated ... much better arch IMHO.